### PR TITLE
feat: add BTC faucet

### PR DESCRIPTION
**Pull request by hstove**
Requires a few ENV variables for RPC client and the faucet private key

### CODE REVIEW

File: src/types/bitcoin-core.d.ts [hunk 1]
```diff
@@ -0,0 +1,1099 @@
+/* eslint-disable tsdoc/syntax */
+declare module 'bitcoin-core' {
```
Comment (zone117x) [inline src/types/bitcoin-core.d.ts:2]: Pulling in this typedef is kinda crazy. It looks like this lib is hostile towards typescript https://github.com/ruimarinho/bitcoin-core/issues?q=typescript

Would you be open to using a different one like https://github.com/vansergen/rpc-bitcoin ?

Comment (hstove) [reply src/types/bitcoin-core.d.ts]: Great suggestion - I looked around for a sec but didn't see that package. Much better. I've implemented it.

File: src/tests/faucet-tests.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import * as RpcClient from 'bitcoin-core';
```
Comment (zone117x) [inline src/tests/faucet-tests.ts:1]: Bad import here

**Comment by hstove** [comment]
@zone117x thanks for the catch, fixed and rebased

**Pull request merged**